deploy-tool/cert/k8_storage.go

```diff
@@ -0,0 +1,151 @@
+package cert
+
+import (
+	"context"
+	"fmt"
+	"github.com/caddyserver/certmagic"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"regexp"
+	"strings"
+)
+
+type SecretStorage struct {
```

Comment:
`KubernetesSecretStorage` seems like a better name to me. `Secret` is the thing we're storing, but as it is the name doesn't give any insight into _where_ we store the secrets.